src/RewardsStreamerMP.sol

```diff
@@ -434,6 +436,21 @@ contract RewardsStreamerMP is
         return _vaultShares(vaultAddress);
     }
 
+    function compound() external onlyNotEmergencyMode {
```

Comment:
This misses `onlyTrustedCodehash` modifier.
We could also add `onlyRegisteredVault` but it's less critical because it's already attached to `stake()`.

So if you come with a vault that isn't registered, you won't be able to stake, then calling `compound()` will be essentially a noop.

Reply:
Ah no, actually it will revert because it has a `0` check.

Reply:
@0x-r4bbit  actually I think we should add a vaultAddress param and let anyone call compound